calyx/src/ir/control.rs

```diff
@@ -80,7 +80,7 @@ pub struct Invoke {
     /// Optional combinational group that is active when the invoke is active.
     pub comb_group: Option<RRC<CombGroup>>,
     /// Mapping from name of external cell in 'comp' to the cell connected to it.
-    pub external_cells: RRC<CellMap>,
+    pub external_cells: Option<CellMap>,
```

Comment:
Is CellMap a hash map? If so, we don’t need to make it an RRC or optional. An empty hashmap just represents an absence of external cells.

Reply:
Cellmap is a vector, same as PortMap. I have changed the code so that external_cells is now a CellMap instead of Option<CellMap>